dev/docker-entrypoint.sh

```diff
@@ -55,10 +55,25 @@ if [ "$1" = '/opt/couchdb/bin/couchdb' ]; then
 	# Ensure that CouchDB will write custom settings in this file
 	touch /opt/couchdb/etc/local.d/docker.ini
 
-	if [ "$COUCHDB_USER" ] && [ "$COUCHDB_PASSWORD" ]; then
-		# Create admin only if not already present
-		if ! grep -Pzoqr "\[admins\]\n$COUCHDB_USER =" /opt/couchdb/etc/local.d/*.ini; then
+
+	# Check if admin users are missing then
+	if ! grep -Pzoqr "\[admins\]\n$COUCHDB_USER =" /opt/couchdb/etc/local.d/*.ini; then
+		# Check if admin credentials provided
+		if [ "$COUCHDB_USER" ] && [ "$COUCHDB_PASSWORD" ]; then
```

Comment:
So, embarrassing admission time, I'd forgotten I'd already added a check in here. We may not need this PR :wink: sorry about that :disappointed: 

This proposed change means that if there's already an admin specified in the file, we don't use `COUCHDB_USER` and `COUCHDB_PASSWORD` at all. That might be surprising to someone. For instance, what if I'm trying to recover a lost password in a stopped container? I can't decide if this is a valid use case or not. What do you think?

Reply:
You're right - my bad. I should have understood what's happening there.. :disappointed:

Containers should be idempotent. When researching if you really leave the password unhashed in the ini file and if your Ini-parser supports any kind of escape-statements for values i stumbled over [Issue 2188: Admin username with = in name gets corrupted on initial password encrypt](https://github.com/apache/couchdb/issues/2188) (where i left an educated guess about the cause for the :bug:).

And i had to learn you're in fact writing the hashed password back into the config file.

So if the password isn't used for anything else then authentication a user should be fine with throwing away the container with the forgotten password and creating a new one (given the data is mounted as external storage).

If the password is also used to internally encrypt or sign anything this wouldn't work though.

Since i guess the first applies you might want to close this PR.

Reply:
Right, you'd have to attach your data to a new container with a new .ini file, server admins aren't stored anywhere other than in the ini files.

We'll go ahead and close this thanks :)